### ISSUE TITLE
Create a custom Policy class

### ISSUE THREAD

**Issue by thiagopbueno** [thiagopbueno/model-aware-policy-optimization#2]
Create a custom `PolicyGraph` that receives a configuration dict and defines actions to take, a trajectory postprocessor for experiences, and a loss function.

**Comment by thiagopbueno** [comment]
Should we extend a [TFPolicy](https://github.com/ray-project/ray/blob/54d5969ceacabad2741af278ef9ed4db0af66835/python/ray/rllib/policy/tf_policy.py#L27) class or call the [build_tf_policy](https://github.com/ray-project/ray/blob/54d5969ceacabad2741af278ef9ed4db0af66835/python/ray/rllib/policy/tf_policy_template.py#L13) helper function with appropriate arguments?

**Comment by angelolovatto** [comment]
@thiagopbueno I would personally prefer to use [build_tf_policy](https://github.com/ray-project/ray/blob/54d5969ceacabad2741af278ef9ed4db0af66835/python/ray/rllib/policy/tf_policy_template.py#L13) for this, as it seems to be simpler and I also believe that it separates each component of the algorithm pretty nicely.

Unfortunately DDPG has not been ported to the new `build_tf_policy` pattern yet, but we can base ourselves off of the [DQN implementation](https://github.com/ray-project/ray/blob/ray-0.7.2/python/ray/rllib/agents/dqn/dqn_policy.py) in the latest release of 'ray'.

**Comment by angelolovatto** [comment]
Renamed the Issue since `PolicyGraph` was renamed to `Policy`

**Comment by angelolovatto** [comment]
I'm considering extending `Policy` directly and organizing it similarly to [keras_policy](https://github.com/ray-project/ray/blob/master/python/ray/rllib/keras_policy.py) since we have three components, each with its own sub-loop for optimization. 

`TFPolicy` expects a single tensor loss. Thus, we would have to summarize all updates in one operation. TD3 actually manages to update the actor every 2 updates of the critic by doing the following:
```python
    @override(TFPolicy)
    def build_apply_op(self, optimizer, grads_and_vars):
        # for policy gradient, update policy net one time v.s.
        # update critic net `policy_delay` time(s)
        should_apply_actor_opt = tf.equal(
            tf.mod(self.global_step, self.config["policy_delay"]), 0)

        def make_apply_op():
            return self._actor_optimizer.apply_gradients(
                self._actor_grads_and_vars)

        actor_op = tf.cond(
            should_apply_actor_opt,
            true_fn=make_apply_op,
            false_fn=lambda: tf.no_op())
        critic_op = self._critic_optimizer.apply_gradients(
            self._critic_grads_and_vars)
        # increment global step & apply ops
        with tf.control_dependencies([tf.assign_add(self.global_step, 1)]):
            return tf.group(actor_op, critic_op)
```
Thus, we could still leverage the `TFPolicy` class if we did something similar to control how often each component has to be updated

**Comment by thiagopbueno** [comment]
Please note that this code might solve issue #12 .

**Comment by angelolovatto** [comment]
PR #19 added our custom policy class, but I forgot to [close this issue by using keywords](https://help.github.com/en/articles/closing-issues-using-keywords), therefore I'm closing it now.
